eval.py

```diff
@@ -75,11 +77,24 @@ def remove_punc(text: str) -> str:
     extra_punct = r"""„“«»¡¿《》！？｡。＂＃＄％＆＇（）＊＋，－／：；＜＝＞＠［＼］＾＿｀｛｜｝～｟｠｢｣､、〃》「」『』【】〔〕〖〗〘〙〚〛〜〝〞〟〰〾〿–—‘’‛“”„‟…‧﹏."""
     punct = punct + extra_punct
     exclude = set(punct)
-    return "".join(ch for ch in text if ch not in exclude)
+    prediction = "".join(ch for ch in text if ch not in exclude)
+    # Now, try harder to strip punctuation using unicode Px prefix
+    # https://stackoverflow.com/questions/48496869/python3-remove-arabic-punctuation/48499154#48499154
+    prediction = "".join(c for c in prediction if not ud.category(c).startswith("P"))
+    return prediction
 
 
 def normalize_answer(text: str) -> str:
-    return white_space_fix(remove_articles(remove_punc(text.lower())))
+    prediction = white_space_fix(remove_articles(remove_punc(text.lower())))
+    # Get rid of some common reasoning corrupted answers:
+    if " answer is " in prediction:
+        prediction = prediction.split(" answer is ")[0]
```

Comment:
While it works for this specific case, I do think that Ruslan was sorta correct that this should be 

```
prediction.split(" answer is ")[-1]
```

To make sure that if there is some text before the model says "the answer is", this line still fetches the part *after* it.